GTBitsOfGood/georgiaCORE
# Justin/persist tree

## Persist Tree Data Structure

- Created mongoose schema for Question
- We now store in a mongodb collection, a bunch of Question documents, questions and options use IDs generated on the client
- Path of making database changes is src/actions ==> pages/api ==> server/mongodb/actions ==> server/mongodb/models
- Added ability to get all questions at once from database and the ability to set all questions at once (I do this by dropping the collection and adding in the new records every time)
- Fixed some UI operations so they update the internal tree correctly, like adding a new node and deleting
- Added a Save button to save current tree
- If the database doesn't have a tree currently, I include a default tree in NavigatorEditor/question.js

Issue Number(s): #8 



### Checklist

- [ x ]  Database schema docs have been updated or are not necessary
- [ x ]  Code follows design and style guidelines
- [ x ]  Code is commented with doc blocks
- [  ]  Tests have been written and executed or are not necessary (still unsure how we are organizing tests)
- [ x ]  Latest code has been rebased from base branch (usually `develop`)
- [ x ]  Commits follow guidelines (concise, squashed, etc)
- [ x ]  Github issues have been linked in relevant commits
- [ x ]  Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes
None

### Related PRs
#24 Should be merged in first since this PR was built on top of the changes in that one

### How to Test
- Go to the *MongoDB* section of this repo's README and follow along
- Start MongoDB instance and change env var filename to .env.local as README says
- Visit /navigation-editor and attempt to create a new edge, delete(backspace), remove options, add options
- Make sure you click Save when you want to save!
